packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -231,7 +231,15 @@ Future<String?> _getCodeSigningIdentityDevelopmentTeam({
 
   final Process opensslProcess = await processUtils.start(
     const <String>['openssl', 'x509', '-subject']);
-  await (opensslProcess.stdin..write(signingCertificateStdout)).close();
+
+  await ProcessUtils.writeToStdinGuarded(
+    stdin: opensslProcess.stdin,
+    line: signingCertificateStdout,
+    onError: (Object? error, _) {
+      throwToolExit('Failed to write to openssl process. Error: $error');
```

Comment:
Is this actionable by the user? We usually throw a tool exit if either the user provided invalid inputs or the state of their machine is in some bad state. This might legitimately be a bug in tool assumptions though, right?

Also, I wonder if the fact that we were synchronously closing this after we call `.write()` (but maybe before the write happened) is why this was crashing in the first place.

Reply:
> Is this actionable by the user? We usually throw a tool exit if either the user provided invalid inputs or the state of their machine is in some bad state. This might legitimately be a bug in tool assumptions though, right?

Great catch. I think I went on autopilot here. I've since replaced this with an `Exception` ([code](https://github.com/flutter/flutter/pull/150120/commits/69452eb6a34cf5f2f12e9d6698959608cff9e429#diff-e24013dd662b438105ef0c1d965a955e4344a6481bc2e6cbe87a30357314fb69R239)).

> Also, I wonder if the fact that we were synchronously closing this after we call .write() (but maybe before the write happened) is why this was crashing in the first place. If that is the case, this change should solve that.

This is certainly possible. As you pointed out, `Process.writeToStdinGuarded` should protect us from this going forward.